addons/pos_self_order/static/src/kiosk/pages/product/product.js

```diff
@@ -49,15 +49,24 @@ export class Product extends Component {
         return increase ? this.state.qty++ : this.state.qty--;
     }
 
-    get fullProductName() {
-        const productAttributeString = Object.values(this.state.selectedVariants).join(", ");
-        let name = `${this.product.name}`;
+    get flattenSelectedAttribute() {
+        return Object.values(this.state.selectedVariants)
+            .map((value) => (value instanceof Set ? Array.from(value) : [parseInt(value)]))
+            .flat();
+    }
 
-        if (productAttributeString) {
-            name += ` (${productAttributeString})`;
+    get fullProductName() {
+        if (!this.product.attributes.length) {
+            return this.product.name;
         }
 
-        return name;
+        const productAttribute = this.product.attributes.map((a) => a.values).flat();
+        const selectedAttributeString = this.flattenSelectedAttribute
+            .map((attributeId) => productAttribute.find((a) => a.id == attributeId).name)
+            .sort() // we need to keep same order each time to be able to compare
+            .join(", ");
+
+        return `${this.product.name} (${selectedAttributeString})`;
```

Comment:
this function is duplicated in `mobile/pages/product_main_view/product_main_view.js`. Maybe you can move it to the product model